src/shared/find-esp.c

```diff
@@ -177,7 +177,8 @@ static int verify_esp_udev(
 
         r = sd_device_get_property_value(d, "ID_PART_ENTRY_SCHEME", &v);
         if (r < 0)
-                return log_error_errno(r, "Failed to get device property: %m");
+                return log_full_errno(searching && r == -ENOENT ? LOG_DEBUG : LOG_ERR, r,
```

Comment:
hmm I think we should return EADDRNOTAVAIL in case `searching && r == -ENOENT` holds, instead of r. After all, if ID_PART_ENTRY_SCHEME is not set it's not the partition we are searching for.